staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -43,14 +43,18 @@ const (
 	userAgentTruncateSuffix = "...TRUNCATED"
 )
 
-func NewEventFromRequest(req *http.Request, requestReceivedTimestamp time.Time, level auditinternal.Level, attribs authorizer.Attributes) (*auditinternal.Event, error) {
-	ev := &auditinternal.Event{
-		RequestReceivedTimestamp: metav1.NewMicroTime(requestReceivedTimestamp),
-		Verb:                     attribs.GetVerb(),
-		RequestURI:               req.URL.RequestURI(),
-		UserAgent:                maybeTruncateUserAgent(req),
-		Level:                    level,
+func LogRequestMetadata(ctx context.Context, req *http.Request, requestReceivedTimestamp time.Time, level auditinternal.Level, attribs authorizer.Attributes) {
```

Comment:
We use the audit package for or [openshift/oauth-server](https://github.com/openshift/oauth-server) and the removal of `NewEventFromRequest` would break our tests. So I rewrote them against your PR and the usage of the audit package is now really nice. All the changes around the removal of `*[]annotations` is amazing! 🎉